src/com/googlecode/jmxtrans/model/output/GangliaWriter.java

```diff
@@ -166,11 +169,14 @@ public void internalWrite(Server server, Query query, ImmutableList<Result> resu
 			if (result.getValues() != null) {
 				for (final Map.Entry<String, Object> resultValue : result.getValues().entrySet()) {
 					final String name = KeyUtils.getKeyString(query, result, resultValue, getTypeNames());
-					final String value = resultValue.getValue().toString();
+					
+					BigDecimal value = new BigDecimal(resultValue.getValue().toString());
+					String strValue = value.compareTo(GANGLIA_MIN_DOUBLE) <= 0 ? "0" : value.toString();
```

Comment:
As what we are doing here is transforming values, we should probably reuse the concept of [ValueTransformer](https://github.com/jmxtrans/jmxtrans/blob/master/src/com/googlecode/jmxtrans/model/results/ValueTransformer.java). We will probably have the same problematic with other output writers. It also makes it much easier to test.